internal/plugin/connectors/tcp/mssql/authentication_handshake.go

```diff
@@ -0,0 +1,132 @@
+package mssql
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"sort"
+
+	mssql "github.com/denisenkom/go-mssqldb"
+)
+
+// PerformFakeHandshakeForClient carries out a fake handshake with a client
+// 1. reads the prelogin packet
+// 2. writes the prelogin repsonse declaring NO support for encryption
+// 3. reads the login packet
+// 4. writes the login response as a success. TODO: what about failure ?
+func PerformFakeHandshakeForClient(clientConn net.Conn) error {
+	// using the default packet size of 4096 (see go-mssqldb/conn_str.go)
+	clientBuf := mssql.NewTdsBuffer(4096, clientConn)
+	msg, err := ReadClientPrelogin(clientBuf)
+	if err != nil {
+		return fmt.Errorf("bad prelogin: %s", err)
+	}
+
+	// based on wireshark, the prelogin response from the server is almost identical
+	// to the login message from the client, with minor changes as below:
+	msg[mssql.PreloginTHREADID] = []byte{}
+	msg[mssql.PreloginVERSION] = []byte{0x0e, 0x00, 0x0c, 0xa6, 0x00, 0x00}
```

Comment:
It would be good to comment on what this binary string is composed of in terms of fields or its string value